### PR TITLE
Feature/maps

### PR DESCRIPTION
**Pull request by nathanhhughes**
@Schmluk about 90% there on the maps; just need to actually take care of printing keys instead of indices (and cleaning up some unintended changes). Will pick this up and finalize it tomorrow morning

### CODE REVIEW

**Comment by nathanhhughes** [comment]
@Schmluk should be ready. I opted to rework how the ADL dispatch works, which lets us now define specializations before `declare_config` is actually defined (which is useful for stl types). I think this is cleaner (it lets us directly use `setValues` and `getValues` on vectors and maps), but lmk if you feel different

File: config_utilities/src/asl_formatter.cpp [hunk 1]
```diff
@@ -63,9 +63,12 @@ std::string AslFormatter::formatErrorsRecursive(const MetaData& data, const std:
   const std::string name_prefix_before = name_prefix_;
   if (Settings::instance().inline_subconfig_field_names) {
     if (!data.field_name.empty()) {
+      // TOOD(nathan) refactor to put in metadata
```
Comment (Schmluk) [inline config_utilities/src/asl_formatter.cpp:66]: I think this is purely a printing problem? Not sure if this needs to go into meta data.

File: config_utilities/test/tests/config_arrays.cpp [hunk 2]
```diff
@@ -47,7 +47,9 @@
 
 namespace config::test {
 
+namespace {
```
Comment (Schmluk) [inline config_utilities/test/tests/config_arrays.cpp:50]: Just for my understanding, does this hide it from other tests or what is the idea here?

**Review by Schmluk — APPROVED** [review]
Looks great! I'm fine with moving the ADL to the other namespace, don't think it makes a difference for any of my purposes.

File: config_utilities/src/asl_formatter.cpp (hunk 1 above)
Comment (nathanhhughes) [reply config_utilities/src/asl_formatter.cpp:66]: yeah, I wasn't sure what the right split between formatting the indices/keys and just getting the string values of the indices/keys if they exist was so I punted on figuring it out for now

**Pull request merged**